examples/DGmethods/ex_001_dycoms.jl

```diff
@@ -58,133 +60,507 @@ eprint = {https://doi.org/10.1175/MWR2930.1}
+            S[k,ev,18] += (localQ[ijk,4,e]/localQ[ijk,1,e] - Horzavgstot[k,ev,4] / Horzavgstot[k,ev,1]) * (thermoQ[ijk,5,e] - Horzavgstot[k,ev,5])
+          end
+        end
+      end
+    end
+  end
+
+#See Outputs below for what S[k,ev,:] are respectively.
+
+  S_avg = zeros(Nqk,nvertelem,18)
+  for s in 1:18
+    for ev in 1:nvertelem
+      for k in 1:Nqk
+        S[k,ev,s] = S[k,ev,s] /  (Nq * Nq * nhorzelem)
+        S_avg[k,ev,s] = MPI.Reduce(S[k,ev,s], +, 0, MPI.COMM_WORLD)
+     
+        if mpirank == 0
+          S_avg[k,ev,s] = S_avg[k,ev,s] / (nranks)
+        end
+      end
+    end
+  end
+
+  OutputWtheta = zeros(nvertelem * Nqk)
+  OutputWQVAP = zeros(nvertelem * Nqk)
+  OutputWU = zeros(nvertelem * Nqk)
+  OutputWV = zeros(nvertelem * Nqk)
+  OutputWW = zeros(nvertelem * Nqk)
+  OutputWRHO = zeros(nvertelem * Nqk)
+  OutputQLIQ = zeros(nvertelem * Nqk)
+  OutputWQLIQ = zeros(nvertelem * Nqk)
+  OutputWWW = zeros(nvertelem * Nqk )
+  OutputUU = zeros(nvertelem * Nqk )
+  OutputVV = zeros(nvertelem * Nqk )
+  OutputZ = zeros(nvertelem * Nqk)
+  OutputU = zeros(nvertelem * Nqk)
+  OutputV = zeros(nvertelem * Nqk)
+  Outputtheta = zeros(nvertelem * Nqk)
+  Outputqt = zeros(nvertelem * Nqk)
+  OutputWqt = zeros(nvertelem * Nqk)
+  Outputthetaliq = zeros(nvertelem * Nqk)
+  OutputWthetav = zeros(nvertelem * Nqk)
+  OutputTKE = zeros(nvertelem * Nqk)
+  OutputWthetal = zeros(nvertelem * Nqk)
+  
+  for ev in 1:nvertelem
+    for k in 1:Nqk
+      i=k + Nqk * (ev - 1)
+      OutputWtheta[i] = S_avg[k,ev,1] # <w'theta'>
+      OutputWQVAP[i] = S_avg[k,ev,2] # <w'qvap'>
+      OutputWU[i] = S_avg[k,ev,3] # <w'u'>
+      OutputWV[i] = S_avg[k,ev,4] # <w'v'>
+      OutputWW[i] = S_avg[k,ev,5] # <w'w'>
+      OutputWRHO[i] = S_avg[k,ev,6] #<w'rho'>
+      OutputQLIQ[i] = S_avg[k,ev,7] # qliq 
+      OutputWQLIQ[i] = S_avg[k,ev,8] # <w'qliq'>
+      OutputWWW[i] = S_avg[k,ev,9]  #<w'w'w'>
+      OutputUU[i] = S_avg[k,ev,10] #<u'u'>
+      OutputVV[i] = S_avg[k,ev,11] #<v'v'>
+      OutputZ[i] = Zvals[k,ev] # Height
+      OutputU[i] =  Horzavgstot[k,ev,2] / Horzavgstot[k,ev,1] # <u> 
+      OutputV[i] =  Horzavgstot[k,ev,3] / Horzavgstot[k,ev,1] # <v>
+      Outputtheta[i] = S_avg[k,ev,12] # <theta> 
+      Outputqt[i] = S_avg[k,ev,13]  # <qt>
+      OutputWqt[i] = S_avg[k,ev,14] #<w'qt'> 
+      Outputthetaliq[i] = S_avg[k,ev,15] # <thetaliq>
+      OutputWthetav[i] = S_avg[k,ev,16] # <w'thetav'>
+      OutputTKE[i] = S_avg[k,ev,17] # <TKE>
+      OutputWthetal[i] = S_avg[k,ev,18] # <w'thetal'>
+    end
+  end
+if mpirank == 0
+
+  io = open(diagnostics_fileout, "a")
+     current_time_str = string(current_time_string, "\n")
+     write(io, current_time_str)
+     writedlm(io, [OutputWtheta OutputWQVAP OutputWU OutputWV OutputWW OutputWRHO OutputQLIQ OutputWQLIQ OutputWWW OutputUU OutputVV OutputU OutputV Outputtheta Outputqt OutputWqt Outputthetaliq OutputWthetav OutputTKE OutputWthetal OutputZ])
```

Comment:
* The columns should have some order. E.g. all velocity moments together, all thermal fluxes together, etc. 
* The current file contains are 20 headers but 21 columns. Please fix.